src/datanode/src/instance/sql.rs

```diff
@@ -12,26 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::Arc;
+
 use async_trait::async_trait;
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
 use common_error::prelude::BoxedError;
 use common_query::Output;
+use common_recordbatch::RecordBatches;
 use common_telemetry::logging::{error, info};
 use common_telemetry::timer;
 use servers::query_handler::SqlQueryHandler;
+use session::context::SessionContext;
 use snafu::prelude::*;
+use sql::ast::ObjectName;
 use sql::statements::statement::Statement;
+use table::engine::TableReference;
 use table::requests::CreateDatabaseRequest;
 
-use crate::error::{
-    BumpTableIdSnafu, CatalogNotFoundSnafu, CatalogSnafu, ExecuteSqlSnafu, ParseSqlSnafu, Result,
-    SchemaNotFoundSnafu, TableIdProviderNotFoundSnafu,
-};
+use crate::error::{self, BumpTableIdSnafu, ExecuteSqlSnafu, Result, TableIdProviderNotFoundSnafu};
 use crate::instance::Instance;
 use crate::metric;
 use crate::sql::SqlRequest;
 
 impl Instance {
-    pub async fn execute_sql(&self, sql: &str) -> Result<Output> {
+    pub async fn execute_sql(&self, sql: &str, session_ctx: Arc<SessionContext>) -> Result<Output> {
```

Comment:
What about adding a `SessionContextRef`? `Arc<SessionContext>` is almost everywhere.

```rust
pub type SessionContextRef = Arc<SessionContext>;
```